File: 02 백준/08 정렬/Python/Prob5648.py

"""
title  : 5648. 역원소 정렬 (Silver 5)
time   : 56ms        136ms
memory : 32140KB     110424KB
"""

import sys

def solution():
    input = sys.stdin.readline

    numbers = list(input().rstrip().split())

    N = int(numbers[0])
    length = len(numbers) - 1

    while length < N:

        temp = list(input().rstrip().split())
        length += len(temp)

        numbers.extend(temp)

    answer = []

    for num in numbers[1:]:

        temp = ""
        idx = len(num) - 1
        
        while idx >= 0:
            temp += num[idx]
            idx -= 1

        answer.append(int(temp))

    for ans in sorted(answer):
        print(ans)

solution()

File: 02 백준/08 정렬/Java/src/Prob5648.java

/**
* title  : 5648. 역원소 정렬 (Silver 5)
* time   : 208ms
* memory : 19748KB
*/

import java.io.*;
import java.util.*;

public class Prob5648 {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        StringTokenizer st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());

        Long[] answer = new Long[N];

        int idx = 0;
        while (true) {

            while (st.hasMoreTokens()) {
                StringBuilder temp = new StringBuilder(st.nextToken());
                answer[idx] = Long.valueOf(String.valueOf(temp.reverse()));
                idx++;
            }

            if (idx == N) {
                break;
            }

            st = new StringTokenizer(br.readLine());
        }

        Arrays.sort(answer);

        for (int i = 0; i < N; i++) {
            sb.append(answer[i]).append("\n");
        }

        System.out.println(sb);
    }
}